Marlin/Marlin_main.cpp

```diff
@@ -3261,7 +3261,7 @@ inline void gcode_M42() {
 
     for (uint8_t i = 0; i < COUNT(sensitive_pins); i++) {
       if (sensitive_pins[i] == pin_number) {
-        pin_number = -1;
+        if (!(Cartridge__GetAugerEnabled && pin_number == SOL0_PIN)) pin_number = -1;
```

Comment:
I'd put `pin_number = -1;` on the next line indented for readability, but that is also just personal preference.